reconcile/database_access_manager.py

```diff
@@ -105,8 +122,13 @@ def _generate_db_access(self) -> str:
         return "".join(statements)
 
     def generate_script(self) -> str:
-        x = self._generate_create_user() + "\n" + self._generate_db_access()
-        return x
+        if not self.db_access.delete:
+            script = self._generate_create_user() + "\n" + self._generate_db_access()
+        else:
+            script = (
+                self._generate_revoke_db_access() + "\n" + self._generate_delete_user()
+            )
+        return script
```

Comment:
Now this method has 2 responsibilities `create` and `delete`, maybe make it explicit by extracting a provision script and a deprovision script like:

```python
    def provision_script(self) -> str:
        return self._generate_create_user() + "\n" + self._generate_db_access()

    def deprovision_script(self) -> str:
        return self._generate_revoke_db_access() + "\n" + self._generate_delete_user()

    def generate_script(self) -> str:
        return (
            self.deprovision_script()
            if self.db_access.delete
            else self.provision_script()
        )
```

Reply:
:+1: